pkg/ruler/rules/store.go

```diff
@@ -21,8 +21,14 @@ var (
 
 // RuleStore is used to store and retrieve rules
 type RuleStore interface {
-	ListAllRuleGroups(ctx context.Context) (map[string]RuleGroupList, error)
```

Comment:
I found it misleading for `List*` methods to also do loading, so I've renamed them and added comment too.

Once this PR is in, I plan to split Listing and Loading into separate methods, because right now Ruler loads all rule groups, even if it's going to throw them right away (due to sharding).

Reply:
Okay, makes sense.